packages/pass-style/src/types.d.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable no-use-before-define */
+import { PASS_STYLE } from './passStyle-helpers.js';
+
+export type PrimitiveStyle =
+  | 'undefined'
+  | 'null'
+  | 'boolean'
+  | 'number'
+  | 'bigint'
+  | 'string'
+  | 'symbol';
+export type ContainerStyle = 'copyRecord' | 'copyArray' | 'tagged';
+export type PassStyle =
+  | PrimitiveStyle
+  | ContainerStyle
+  | 'remotable'
+  | 'error'
+  | 'promise';
+export type PassStyled<S extends string> = {
+  [PASS_STYLE]: S;
+};
+export type ExtractStyle<P extends PassStyled<any>> = P[typeof PASS_STYLE];
+export type PassByCopy =
+  | import('type-fest').Primitive
+  | Error
+  | any[]
+  | CopyRecord
```

Comment:
Curious why we do use `CopyRecord` rather than `Record` but we use `any[]` rather than `CopyArray`?

Also, shouldn't this at least be `Passable[]`, or does this encounter recursive type problems again?

Reply:
changed to CopyArray